src/UniversalRewardsDistributor.sol

```diff
@@ -95,12 +96,17 @@ contract UniversalRewardsDistributor is IUniversalRewardsDistributor {
         require(pendingRoot.submittedAt > 0, ErrorsLib.NO_PENDING_ROOT);
         require(block.timestamp >= pendingRoot.submittedAt + timelock, ErrorsLib.TIMELOCK_NOT_EXPIRED);
 
-        root = pendingRoot.root;
-        ipfsHash = pendingRoot.ipfsHash;
+        _setRoot(pendingRoot.root, pendingRoot.ipfsHash);
+    }
 
-        emit EventsLib.RootSet(pendingRoot.root, pendingRoot.ipfsHash);
+    /// @notice Revokes the pending root.
+    /// @dev Can be frontrunned with `acceptRoot` in case the timelock has passed.
```

Comment:
I really think this kind of comment is valuable, unfortunately it is inconsistent because there are other function which i suspect could be frontrunned by others
Let's keep it but I think we should dedicate resources in documenting these later